clippy_lints/src/misc.rs

```diff
@@ -224,7 +244,23 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                                                initref=initref));
                 }
             );
-        }}
+        }};
+        if_let_chain! {[
+            let StmtSemi(ref expr, _) = s.node,
+            let Expr_::ExprBinary(ref binop, ref a, ref b) = expr.node,
+            binop.node == BiAnd || binop.node == BiOr,
+            let Some(sugg) = Sugg::hir_opt(cx, a),
```

Comment:
We already have an in_macro check, so it should mostly work as it is, but some fancy use of macros might end up not triggering the lint. But this is probably true for a lot of lints, so don't worry about it. It's only a false negative in that case. Better to err on this side than to annoy with wrong lints